preload/prefetch-document.html

```diff
@@ -0,0 +1,104 @@
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+<script src="/common/dispatcher/dispatcher.js"></script>
+<script src="resources/prefetch-helper.js"></script>
+<body>
+<script>
+
+const {ORIGIN, REMOTE_ORIGIN, HTTP_NOTSAMESITE_ORIGIN} = get_host_info();
+const origins = {
+    "same origin": ORIGIN,
+    "same site": REMOTE_ORIGIN,
+    "different site": HTTP_NOTSAMESITE_ORIGIN
+}
+const loaders = {
+    image: {
+        file: '../../images/green.png',
+        type: 'image/png',
+        load: href => {
+            const image = document.createElement('img');
+            image.src = href;
+            document.body.appendChild(image);
+            return new Promise(resolve => image.addEventListener('load', resolve));
+        }
+    },
+    script: {
+        file: 'dummy.js',
+        type: 'application/javascript',
+        load: href => {
+            const script = document.createElement('script');
+            script.src = href;
+            document.body.appendChild(script);
+            return new Promise(resolve => script.addEventListener('load', resolve));
+        }
+    },
+    style: {
+        file: 'dummy.css',
+        type: 'text/css',
+        load: href => {
+            const link = document.createElement('link');
+            link.href = href;
+            link.rel = "stylesheet";
+            document.body.appendChild(link);
+            return new Promise(resolve => link.addEventListener('load', resolve));
+        }
+    },
+    document: {
+        file: 'empty.html',
+        type: 'text/html',
+        load: href => {
+            const iframe = document.createElement("iframe");
+            iframe.src = href;
+            document.body.appendChild(iframe);
+            return new Promise(resolve => iframe.addEventListener("load", resolve));
+        }
+    }
+};
+
+async function prefetch_document(origin, as, t) {
+    const {href, uid} = await prefetch({
+        file: "prefetch-exec.html",
+        type: "text/html",
+        corssOrigin: "anonymous",
+        origin: origins[origin], as});
+    const popup = window.open(href);
+    const remoteContext = new RemoteContext(uid);
+    t.add_cleanup(() => popup.close());
+    const result = await remoteContext.execute_script(() => "OK");
+    assert_equals(result, "OK");
+    return await get_prefetch_info(href);
+}
+
+async function test_prefetch_document({origin, as}, expected) {
+    promise_test(async t => {
+        const requests = await prefetch_document(origin, as, t);
+        const did_consume = requests.length === 1 ? "consumed" : "not consumed";
+        assert_equals(did_consume, expected);
+    }, `Prefetching a document (${origin}, as="${as}") should be ${expected}`);
+}
+
+test_prefetch_document({origin: "same origin"}, "consumed");
+test_prefetch_document({origin: "same site"}, "consumed");
```

Comment:
These two will end up setting `link.as = "undefined"`, I believe. Is that intentional? Probably they should not set `as`.